dags/elsevier/repository.py

```diff
@@ -0,0 +1,61 @@
+import os
+from io import BytesIO
+from typing import IO
+
+from common.repository import IRepository
+from common.s3_service import S3Service
+from common.utils import find_extension
+
+
+class ElsevierRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.s3 = S3Service(os.getenv("ELSEVIER_BUCKET_NAME", "elsevier"))
+
+    def get_all_raw_filenames(self):
+        return [
+            f.key.replace("raw/", "")
+            for f in self.s3.objects.filter(Prefix=self.ZIPED_DIR).all()
+        ]
+
+    def find_all(self, filenames_to_process=None):
+        grouped_files = {}
+        filenames = (
+            filenames_to_process
+            if filenames_to_process
+            else self.__find_all_extracted_files()
+        )
+        for file in filenames:
+            last_part = os.path.basename(file)
+            filename_without_extension = last_part.split(".")[0]
+            if filename_without_extension not in grouped_files:
+                grouped_files[filename_without_extension] = {}
+            extension = find_extension(last_part)
+            grouped_files[filename_without_extension][extension] = file
+        return list(grouped_files.values())
+
+    def get_by_id(self, id: str):
+        retfile = BytesIO()
+        self.s3.download_fileobj(id, retfile)
+        return retfile
+
+    def save(self, filename: str, obj: IO):
+        obj.seek(0)
+        prefix = self.ZIPED_DIR if ".tar" in filename else self.EXTRACTED_DIR
+        self.s3.upload_fileobj(obj, prefix + filename)
+
+    def delete_all(self):
+        self.s3.objects.all().delete()
+
+    def __find_all_extracted_files(self):
```

Comment:
why this method has to be private?

Reply:
because it's inner function, it has to be used just in the [find_all](https://github.com/cern-sis/workflows/blob/4b054c906a2f81ae3a818d7a64d1656c996cdfaf/dags/elsevier/repository.py#L29) method:
https://github.com/cern-sis/workflows/blob/4b054c906a2f81ae3a818d7a64d1656c996cdfaf/dags/elsevier/repository.py#L29

Reply:
it's not inner function in python terms ;) inner function is sth like
```
def outer():
    a = []
    def inner():
        a.append(5)
    return inner

```
you can make protected maybe, it's very rare to use private method's in python (unlike in Java, for example). IMO if you go into this direction you should be very consistent about it.